ide/commons-gwt/src/main/java/org/eclipse/che/ide/jsonrpc/impl/NotificationDispatcher.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.jsonrpc.impl;
+
+import com.google.gwt.json.client.JSONObject;
+
+import org.eclipse.che.ide.dto.DtoFactory;
+import org.eclipse.che.ide.jsonrpc.RequestHandler;
+import org.eclipse.che.ide.util.loging.Log;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Map;
+
+/**
+ * Dispatches incoming json rpc notification
+ *
+ * @author Dmitry Kuleshov
+ */
+@Singleton
+public class NotificationDispatcher extends AbstractJsonRpcDispatcher {
+    private final DtoFactory dtoFactory;
+
+    @Inject
+    public NotificationDispatcher(Map<String, RequestHandler> handlers, DtoFactory dtoFactory) {
+        super(handlers);
+        this.dtoFactory = dtoFactory;
+    }
+
+    /**
+     * Dispatches json rpc notification received from endpoint identified by a
+     * high level identifier and represented as a json object.
+     *
+     * @param endpointId high level endpoint identifier
```

Comment:
javadoc isn't formatted with che code style.
Please, review all classes in **org.eclipse.che.ide.jsonrpc.impl** package because several of them have the same problem.